ssr/src/page/profile/mod.rs

```diff
@@ -45,53 +46,69 @@ fn Stat(stat: u64, #[prop(into)] info: String) -> impl IntoView {
         </div>
     }
 }
-
+#[derive(Params, Clone, PartialEq)]
+struct TabsParam {
+    tab: String,
+}
 #[component]
-fn ListSwitcher(user_canister: Principal, user_principal: Principal) -> impl IntoView {
-    let (cur_tab, set_cur_tab) = create_query_signal::<String>("tab");
+fn ListSwitcher1(user_canister: Principal, user_principal: Principal) -> impl IntoView {
+    let param = use_params::<TabsParam>();
+
     let current_tab = create_memo(move |_| {
-        with!(|cur_tab| match cur_tab.as_deref() {
-            Some("posts") => 0,
-            Some("speculations") => 1,
-            Some("tokens") => 2,
-            _ => 0,
+        param.with(|p| match p {
```

Comment:
nit: you can do p.unwrap_or_default() to avoid a nested match

Reply:
This will require us to clone p because, even if we use `p.as_ref().unwrap_or_default()` , the `unwrap_or_default()` won't work. We'd need to implement `Default` for the `&TabsParam` struct, which would involve dealing with lifetimes and other complexities. Additionally, dereferencing can't be used since it's behind a shared reference.

Reply:
i've pushed the fix